### ISSUE TITLE
Add a configuration to pick a branch where the annotated tags bump the minor or major version.

### ISSUE THREAD

**Issue by valenca** [jgitver/jgitver#72]
In the case a project has a master branch from which hotfix versions are spawned, we often run into the following problem:

 1. Commit A in branch master is tagged with version 1.0.0 and released.
 1. A new hotfix branch is created from it, which jgitver evaluates as 1.0.1-SNAPSHOT
 1. The next commit in the master branch receives a new 1.1.0 lightweight tag as it is now the development branch for 1.1.0-SNAPSHOT

In regular development, this works perfectly fine if there is a short interval between steps 1 and 3. But if it is a particularly slow repository, and the first commit in the master branch only comes, say, a month after the release, there is a high chance that the developers will forget to push the new lightweight tag with their commit, and both the master and the hotfix branches will begin evaluating (and possibly deploying) to 1.0.1-SNAPSHOT, which can cause overwriting and confusion.

I propose an improvement/feature where we can pick one (or more) branches where after each annotated tag it defaults to evaluating a bump in the Minor version, as opposed to the patch version.

**Comment by valenca** [comment]
Yes, it looks like jgitver/jgitver-maven-plugin#104 is proposing a different solution to the same problem, overall.

The solution I proposed has as a goal the ability to, after tagging the release, no other commit needs to be done in order to establish non-overlapping versions in the various branches.

Quickly looking at the code, the solution would be something like this, in MaverVersionStrategy.java:53:

    if (GitUtils.isAnnotated(tagToUse) && !baseVersion.removeQualifier("SNAPSHOT").isQualified()) {
        // found tag to use was a non qualified annotated one, lets' increment the version automatically
        if ("master".equals(getRepository().getBranch())){
            baseVersion = baseVersion.incrementMinor();
        } else {
            baseVersion = baseVersion.incrementPatch();
        }
    }

Where "master" would be replaced by the config check, or something.

**Comment by valenca** [comment]
I guess this would also have to apply to any branches coming out of the selected ones for consistency sake, which adds another layer of logic to it.

**Comment by mmiglier** [comment]
Currently we create an empty commit on develop right after creating a release branch just to add a lightweight tag for forcing develop version to increment minor instead of patch.

It would be nice to be able to set autoIncrementMinor on develop.

**Comment by valenca** [comment]
> It would be nice to be able to set autoIncrementMinor on develop.

Yes, but see my previous comment: any branches branching out of develop after that commit should also be able to see that bump (which they wouldn't if it was strictly linked to the branch name).
